NDTensors/ext/NDTensorsCUDAExt/indexing.jl

```diff
@@ -1,8 +1,18 @@
-function Base.getindex(::Type{<:CuArray}, T::DenseTensor{<:Number})
-  return CUDA.@allowscalar data(T)[]
+function Base.getindex(E::Exposed{<:CuArray})
+  return CUDA.@allowscalar unexpose(E)[]
 end
 
-function Base.setindex!(::Type{<:CuArray}, T::DenseTensor{<:Number}, x::Number)
-  CUDA.@allowscalar data(T)[] = x
-  return T
+function setindex!(E::Exposed{<:CuArray}, x::Number)
+  CUDA.@allowscalar unexpose(E)[] = x
+  return unexpose(E)
+end
+
+function Base.getindex(E::Exposed{<:CuArray,<:Adjoint}, I...)
+  Ep = parent(E)
+  return Base.getindex(Ep, I...)
```

Comment:
```suggestion
  Ap = parent(E)
  return expose(Ap)[I...]
```

Reply:
I think the way you wrote it, `Ep` would not be `Exposed` since `parent` would unexpose it.